marconi-chain-index/test/Spec/Marconi/ChainIndex/Indexers/Utxo/UtxoIndex.hs

```diff
@@ -97,72 +84,84 @@ tests = testGroup "Spec.Marconi.ChainIndex.Indexers.Utxo"
+--    - database rerieved Spent TxIns is a function of `depth` and `slotNo`, and thus potentially
+--  A consequence of the above is that the set of databaset fetched `Sptnt TxIns` may not match those of the original event
+propSaveToAndRetrieveFromUtxoStore :: Property
+propSaveToAndRetrieveFromUtxoStore = property $ do
+  let depth = Utxo.Depth 2160 -- we set a high valued for depth, to make sure we only test in-memory buffer
   events <- forAll UtxoGen.genUtxoEvents
   (storedEvents :: [StorableEvent Utxo.UtxoHandle]) <-
-    (liftIO . Utxo.open ":memory:") (Utxo.Depth 10)
+    (liftIO $ Utxo.open ":memory:" depth False) -- don't vacuum sqlite
      >>= liftIO . Storable.insertMany events
      >>= liftIO . Storable.getEvents
-  Hedgehog.assert (equivalentLists storedEvents events)
-
--- Insert Utxo events in storage, and retrieve the events by address
---
--- utxoInsertAndQueryTest :: Property
--- utxoInsertAndQueryTest = property $ do
---   events <- forAll genUtxoEvents
---   depth <- forAll $ Gen.int (Range.linear 1 5)
---   indexer <- liftIO $ Utxo.open ":memory:" (Utxo.Depth depth)
---              >>= liftIO . Storable.insertMany events
---   let
---     qs :: [StorableQuery Utxo.UtxoHandle]
---     qs = fmap (Utxo.UtxoAddress . Utxo._address) . concatMap (Set.toList . Utxo.ueUtxos) $ events
---   results <- liftIO . traverse (Storable.query Storable.QEverything indexer) $ qs
---   let rows = concatMap (\(Utxo.UtxoResult rs) -> rs ) results
---   computedEvent <-
---     liftIO . Utxo.rowsToEvents (Utxo.getTxIns (getConn indexer) ) $ rows
---   Hedgehog.assert (equivalentLists
---                    computedEvent
---                    (filter (\e -> Utxo.ueChainPoint e /= C.ChainPointAtGenesis) events) )
+  List.sort storedEvents === List.sort events
 
 -- Insert Utxo events in storage, and retreive the events by address and query interval
---
-utxoQueryIntervalTest :: Property
-utxoQueryIntervalTest = property $ do
-  event0 <- forAll $ genEventWithShelleyAddressAtChainPoint C.ChainPointAtGenesis
-  event1 <- forAll $ genEventWithShelleyAddressAtChainPoint (head chainpoints)
-  event2 <- forAll $ genEventWithShelleyAddressAtChainPoint (chainpoints !! 1)
-  event3 <- forAll $ genEventWithShelleyAddressAtChainPoint (chainpoints !! 2)
-  let events = [event0, event1, event2, event3]
-  indexer <- liftIO $ Utxo.open ":memory:" (Utxo.Depth 2)
-             >>= liftIO . Storable.insertMany [event0, event1, event2, event3]
-  let
-    qs :: [StorableQuery Utxo.UtxoHandle]
-    qs = fmap (Utxo.UtxoAddress . Utxo._address) . concatMap (Set.toList . Utxo.ueUtxos) $ events
-  results <- liftIO . traverse (Storable.query (Storable.QInterval (head chainpoints)(chainpoints !! 1)) indexer) $ qs
-  let rows = concatMap (\(Utxo.UtxoResult rs) -> rs ) results
-  computedEvent <-
-    liftIO . Utxo.rowsToEvents (Utxo.getTxIns (getConn indexer) ) $ rows
-  Hedgehog.assert (equivalentLists computedEvent [event0, event1])
```

Comment:
@koslambrou this is another example of previously failing tests.  This test was passing since the `equivalentLists` was not really comparing anything